File: iOS/AppFactory/AppFactory/Http/Requester/EstimateRequester.swift
//
//  EstimateRequester.swift
//  AppFactory
//
//  Created by Leapfrog-Software on 2018/04/08.
//  Copyright © 2018年 Leapfrog-Inc. All rights reserved.
//

import Foundation

struct EstimateRequestData {
    let target: String
    let purpose: String
    let description: String
    let ios: Bool
    let android: Bool
    let chat: Bool
    let camera: Bool
    let movie: Bool
    let push: Bool
    let map: Bool
    let geofence: Bool
    let settle: Bool
    let credit: Bool
    let user: Bool
    let sns: Bool
    let notes: String
    let email: String
}

class EstimateRequester {
    
    class func send(requestData: EstimateRequestData, completion: @escaping ((Bool) -> ())) {
        
        var url = Constants.ServerUrl
        url += "?"
        url += ("command=estimate")
        url += "&"
        url += ("target=" + requestData.target)
        url += "&"
        url += ("purchase=" + (requestData.purpose.base64Encode() ?? ""))
        url += "&"
        url += ("description=" + (requestData.description.base64Encode() ?? ""))
        url += "&"
        url += ("ios=" + (requestData.ios ? "1" : "0"))
        url += "&"
        url += ("android=" + (requestData.android ? "1" : "0"))
        url += "&"
        url += ("chat=" + (requestData.chat ? "1" : "0"))
        url += "&"
        url += ("camera=" + (requestData.camera ? "1" : "0"))
        url += "&"
        url += ("movie=" + (requestData.movie ? "1" : "0"))
        url += "&"
        url += ("push=" + (requestData.push ? "1" : "0"))
        url += "&"
        url += ("map=" + (requestData.map ? "1" : "0"))
        url += "&"
        url += ("geofence=" + (requestData.geofence ? "1" : "0"))
        url += "&"
        url += ("settle=" + (requestData.settle ? "1" : "0"))
        url += "&"
        url += ("credit=" + (requestData.credit ? "1" : "0"))
        url += "&"
        url += ("user=" + (requestData.user ? "1" : "0"))
        url += "&"
        url += ("sns=" + (requestData.sns ? "1" : "0"))
        url += "&"
        url += ("notes=" + (requestData.notes.base64Encode() ?? ""))
        url += "&"
        url += ("email=" + (requestData.email.base64Encode() ?? ""))
        
        ApiManager.get(url: url, completion: { result, data in
            if result, let result = data?["result"] as? String, result == "0" {
                completion(true)
            } else {
                completion(false)
            }
        })
    }
}

File: Android/app/src/main/java/leapfrog_inc/appfactory/Http/Requester/EstimateRequester.java
package leapfrog_inc.appfactory.Http.Requester;

import android.util.Base64;

import org.json.JSONObject;

import leapfrog_inc.appfactory.Function.Constants;
import leapfrog_inc.appfactory.Http.HttpManager;

/**
 * Created by Leapfrog-Software on 2018/04/16.
 */

public class EstimateRequester {

    public static class EstimateRequestData {
        public String target;
        public String purpose;
        public String description;
        public boolean ios;
        public boolean android;
        public boolean chat;
        public boolean camera;
        public boolean movie;
        public boolean push;
        public boolean map;
        public boolean geofence;
        public boolean settle;
        public boolean credit;
        public boolean user;
        public boolean sns;
        public String notes;
        public String email;
    }

    public static void send(EstimateRequestData requestData, final EstimateRequesterCallback callback) {

        HttpManager httpManager = new HttpManager(new HttpManager.HttpCallback() {
            @Override
            public void didReceiveData(boolean result, String data) {

                if (result) {
                    try {
                        JSONObject json = new JSONObject(data);
                        if (json.getString("result").equals("0")) {
                            callback.didReceiveData(true);
                            return;
                        }
                    } catch (Exception e) {}
                }
                callback.didReceiveData(false);
            }
        });

        StringBuffer url = new StringBuffer();
        url.append(Constants.ServerApiUrl);
        url.append("?");
        url.append("command=estimate");
        url.append("&");
        url.append(("target=" + requestData.target));
        url.append("&");
        url.append(("purpose=" + new String(Base64.encode(requestData.purpose.getBytes(), Base64.URL_SAFE | Base64.NO_WRAP))));
        url.append("&");
        url.append(("description=" + new String(Base64.encode(requestData.description.getBytes(), Base64.URL_SAFE | Base64.NO_WRAP))));
        url.append("&");
        url.append(("ios=" + (requestData.ios ? "1" : "0")));
        url.append("&");
        url.append(("android=" + (requestData.android ? "1" : "0")));
        url.append("&");
        url.append(("chat=" + (requestData.ios ? "1" : "0")));
        url.append("&");
        url.append(("camera=" + (requestData.ios ? "1" : "0")));
        url.append("&");
        url.append(("movie=" + (requestData.ios ? "1" : "0")));
        url.append("&");
        url.append(("push=" + (requestData.ios ? "1" : "0")));
        url.append("&");
        url.append(("map=" + (requestData.ios ? "1" : "0")));
        url.append("&");
        url.append(("geofence=" + (requestData.ios ? "1" : "0")));
        url.append("&");
        url.append(("settle=" + (requestData.ios ? "1" : "0")));
        url.append("&");
        url.append(("credit=" + (requestData.ios ? "1" : "0")));
        url.append("&");
        url.append(("user=" + (requestData.ios ? "1" : "0")));
        url.append("&");
        url.append(("sns=" + (requestData.ios ? "1" : "0")));
        url.append("&");
        url.append(("notes=" + new String(Base64.encode(requestData.notes.getBytes(), Base64.URL_SAFE | Base64.NO_WRAP))));
        url.append("&");
        url.append(("email=" + new String(Base64.encode(requestData.email.getBytes(), Base64.URL_SAFE | Base64.NO_WRAP))));

        httpManager.execute(url.toString(), "GET", null);
    }

    public interface EstimateRequesterCallback {
        void didReceiveData(boolean result);
    }
}
